### PR TITLE
Use customer and sample filters

### PR DESCRIPTION
**Pull request by seallard**
### Added
- Customer filter EXCLUDE_INTERNAL_ID

### Fixed
- Move _get_filtered_case_query to base handler

### Changed
- Use Customer and Sample filters in _get_filtered_case_query

## Review
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by

### This [version](https://semver.org/) is a

- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/store/api/base.py [hunk 1]
```diff
@@ -127,3 +131,77 @@ def _get_latest_analyses_for_cases_query(self) -> Query:
...
+        name: str,
+        priority: str,
+        sample_id: str,
+    ) -> Query:
+        cases_query: Query = self._get_query(table=Family)
+        filter_functions: List[Callable] = []
+
+        filter_case_order_date = None
+        if days != 0:
+            filter_case_order_date = datetime.datetime.now() - datetime.timedelta(days=days)
```
Comment (henrikstranneheim) [inline cg/store/api/base.py:152]: I think there is a utility function get_date_days_ago for this type of date handling

File: cg/store/api/base.py [hunk 2]
```diff
@@ -127,3 +131,77 @@ def _get_latest_analyses_for_cases_query(self) -> Query:
                 self.Analysis.started_at == case_and_date_subquery.c.started_at,
             ),
         )
+
+    def _get_filtered_case_query(
```
Comment (henrikstranneheim) [inline cg/store/api/base.py:135]: Great work on this one. We should note this function down for a major refactoring in the future

File: cg/store/filters/status_customer_filters.py [hunk 3]
```diff
@@ -13,21 +13,31 @@ def filter_customer_by_customer_internal_id(
     return customers.filter(Customer.internal_id == customer_internal_id)
 
 
+def filter_customer_by_exclude_customer_internal_id(
+    customers: Query, exclude_customer_internal_id: str, **kwargs
+) -> Query:
+    """Return customer by customer internal id."""
```
Comment (henrikstranneheim) [inline cg/store/filters/status_customer_filters.py:19]:
```suggestion
    """Return customer excluded by customer internal id."""
```

**Review by henrikstranneheim — APPROVED** [review]
Well done 👍!

File: cg/store/api/base.py (hunk 2 above)
Comment (seallard) [reply cg/store/api/base.py:135]: Agreed!

**Pull request merged**